src/DynamoCoreWpf/Views/Core/AnnotationView.xaml

```diff
@@ -250,74 +250,188 @@
             <RowDefinition Height="*"></RowDefinition>
         </Grid.RowDefinitions>
         <Grid.ContextMenu>
-            <ContextMenu>
-                <MenuItem  Header="{x:Static p:Resources.GroupContextMenuDeleteGroup}"
-                           Name="DeleteAnnotation"
-                           Click="OnDeleteAnnotation" />
-                <MenuItem  Header="{x:Static p:Resources.GroupContextMenuUngroup}"
-                           Name="UngroupAnnotation"
-                           Click="OnUngroupAnnotation" />
-                <MenuItem Header="{x:Static p:Resources.ContextUnGroupFromSelection}"
-                          Name="unGroup_cm"
-                          Command="{Binding Path=RemoveGroupFromGroupCommand}" />
-                <MenuItem Header="{x:Static p:Resources.GroupContextMenuAddGroupToGroup}"
-                          Name="AddGroupToGroup"
-                          Command="{Binding AddGroupToGroupCommand}" />
-                <MenuItem  Header="{x:Static p:Resources.GroupContextMenuGraphLayout}"
-                           Name="GraphLayoutAnnotation"
-                           Click="OnGraphLayoutAnnotation" />
-                <MenuItem  Header="{x:Static p:Resources.GroupContextMenuFont}"
-                           Name="ChangeSize">
-                    <MenuItem Header="14"
-                              Name="FontSize0"
+            <ContextMenu Background="#666666" Style="{StaticResource ContextMenuStyle}">
+                <ContextMenu.Resources>
+                    <ResourceDictionary>
+                        <Style TargetType="{x:Type MenuItem}">
+                            <Setter Property="IsChecked" Value="{DynamicResource IsChecked}" />
+                            <!--<Setter Property="Height" Value="30" />-->
+                            <Setter Property="Width" Value="240" />
+                            <Setter Property="Padding" Value="20,0,20,0" />
+                            <Setter Property="Template">
```

Comment:
question: do we repeat a lot of the following style definitions?

Reply:
You mean the MenuItem style? 
We do; this is something I've struggled with. 
Because our Menus contain MenuItems, but also Separators (and sometimes other things too, e.g. the Node Group Color ItemCollections), we get an exception when setting the MenuItem style.

The workaround is to have the styles passively sitting around in a ContextMenu.Resources scope, being applied when the correct Type is found. That's what we do current. However, I have issues when trying to define a ContextMenu.Resources scope _inside_ of a ResourceDictionary. So while we can set a global ContextMenu Style just fine, we can't do the same for ContextMenuItems. It would be great if we could - could clean up a lot of code!

Reply:
Thank you for the reasoning, glad this has been considered